### PR TITLE
xrt msgg formatting

### PR DESCRIPTION
**Pull request by AShivangi**
1. Enhanced the messaging system
2. Added capability to pick up .ini from the current directory
3. Implemented verbosity (0-7)
4. Testcase under "XRT/tests/message-mgmt"

The testcase runs 3 times with 3 different sdaccel.ini settings. Verbosity is always set to 7 to show all messages. 
RUN 1: Picks up .ini file from the same folder as exe and logs in "mylog.run" file in that folder
RUN 2: Picks up .ini file from the same folder as exe and shows log output on the console
RUN 3: Picks up .ini file from the the current working directtory and logs in "mylog.run" file in that folder

**Running**:
```
$ cd XRT/tests/message-mgmt
$ ./run.sh #to run all scenarios
$ make clean #to remove .exe and .log files
```

**Output:**
```
RUN 1: Logging in myrun.log                                                                                      
RUN 2: Logging on console                                                                                        
Cannot find index 2                                                                                              
XRT build version: 2.2.0
Build hash: a921cb8936ba347bf1bac885cac5533b08f72e9b
Build date: 2019-02-19 14:57:28
Git branch: xma-msggs
PID: 257118
UID: 31939
[Tue Feb 19 16:55:45 2019]
HOST: xsjbrd2
EXE: /home/sagarw/github/XRT/tests/message-mgmt/host.exe
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647535208256,  EMERGENCY: (0) Unable to open device 2
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647535208256,  INFO: (6) hello.xclbin was passed in as an argument
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647535208256,  DEBUG: (7) Debug flag was set
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647535208256,  ALERT: (1) Unable to obtain device information
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647535208256,  CRITICAL: (2) Unable to load xclbin
~~~Multi threading~~~
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647510935296,  NOTICE: (5) Running Thread number 1
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647502542592,  NOTICE: (5) Running Thread number 2
~~~~~~~~~~~~~~~~~~~~~
Other messages:
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647535208256,  ERROR: (3) Display when verbosity 3
[Tue Feb 19 16:55:45 2019] [XRT] Tid:139647535208256,  WARNING: (4) Display when verbosity 2
RUN 3: Picking .ini from current directory (XRT/tests/) and logging in myrun.log
```

More details: [http://confluence.xilinx.com/display/~sagarw/Message+mgmt](http://confluence.xilinx.com/display/~sagarw/Message+mgmt)

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

**Comment by sonals** [comment]
retest this please

File: src/runtime_src/driver/include/xclhal2.h [hunk 1]
```diff
@@ -496,12 +495,13 @@ XCL_DRIVER_DLLESPEC unsigned int xclVersion();
  *
  * @handle:        Device handle
  * @level:         Severity level of the msg
+ * @tag:           Name of the process giving the message
```
Comment (sonals) [inline src/runtime_src/driver/include/xclhal2.h]: A TAG supplied by the client, like "OCL", "XMA", etc.

File: src/runtime_src/driver/xclng/xrt/util/message.cpp [hunk 2]
```diff
@@ -144,40 +167,86 @@ syslog_dispatch::~syslog_dispatch() {
...
 
 //file ops
 file_dispatch::file_dispatch(const std::string &file) {
   handle.open(file.c_str());
+  handle << "XRT build version: " << xrt_build_version << std::endl;
+  handle << "Build hash: " << xrt_build_version_hash << std::endl;
+  handle << "Build date: " << xrt_build_version_date << std::endl;
+  handle << "Git branch: " << xrt_build_version_branch<< std::endl;
+  //timestamp
+  auto time = std::chrono::system_clock::now();
```
Comment (stsoe) [inline src/runtime_src/driver/xclng/xrt/util/message.cpp]: Lets wrap this chrono call and conversion to string in a single function.  
Copy runtime_src/xrt/util/time.{h,cpp} to xclng/xrt/util.h (change namespace to xrt_core) and add the function you need.

File: src/runtime_src/driver/xclng/xrt/util/message.cpp [hunk 3]
```diff
@@ -144,40 +167,86 @@ syslog_dispatch::~syslog_dispatch() {
...
 file_dispatch::~file_dispatch() {
   handle.close();
 }
 
 void
-file_dispatch::send(severity_level l, const char* msg) {
-  handle << severityMap[l] << msg << std::endl;
+file_dispatch::send(severity_level l, const char* tag, const char* msg) {
+
+  auto time = std::chrono::system_clock::now();
```
Comment (stsoe) [inline src/runtime_src/driver/xclng/xrt/util/message.cpp]: Use new function

File: src/runtime_src/driver/xclng/xrt/util/message.cpp [hunk 4]
```diff
@@ -144,40 +167,86 @@ syslog_dispatch::~syslog_dispatch() {
...
 //console ops
+console_dispatch::console_dispatch() {
+  std::cout << "XRT build version: " << xrt_build_version << std::endl;
+  std::cout << "Build hash: " << xrt_build_version_hash << std::endl;
+  std::cout << "Build date: " << xrt_build_version_date << std::endl;
+  std::cout << "Git branch: " << xrt_build_version_branch<< std::endl;
+  std::cout << "PID: " << getpid() << std::endl;
+  std::cout << "UID: " << getuid() << std::endl;
+  //timestamp
+  auto time = std::chrono::system_clock::now();
```
Comment (stsoe) [inline src/runtime_src/driver/xclng/xrt/util/message.cpp]: use new function

File: src/runtime_src/driver/xclng/xrt/util/message.cpp [hunk 5]
```diff
@@ -144,40 +167,86 @@ syslog_dispatch::~syslog_dispatch() {
...
+  char hostname[HOST_NAME_MAX];
+  gethostname(hostname, HOST_NAME_MAX);
+  std::cout << "HOST: " <<  hostname << std::endl;
+  std::cout << "EXE: " <<get_exe_path() << std::endl;
+}
 void
-console_dispatch::send(severity_level l, const char* msg) {
-  std::cout << severityMap[l]  << msg << std::endl;
+console_dispatch::send(severity_level l, const char* tag, const char* msg) {
+  auto time = std::chrono::system_clock::now();
```
Comment (stsoe) [inline src/runtime_src/driver/xclng/xrt/util/message.cpp]: Use new function

File: src/runtime_src/driver/xclng/xrt/util/message.cpp [hunk 6]
```diff
@@ -44,27 +70,26 @@ class null_dispatch : public message_dispatch
...
 class console_dispatch : public message_dispatch
 {
 public:
-  console_dispatch() {}
+  console_dispatch();
   virtual ~console_dispatch() {}
-  virtual void send(severity_level l, const char* msg) override;
+  virtual void send(severity_level l, const char* tag, const char* msg) override;
 private:
   std::map<severity_level, const char*> severityMap = {
```
Comment (sonals) [inline src/runtime_src/driver/xclng/xrt/util/message.cpp]: constexpr static std::map<severity_level, const char*> severityMap = {

File: src/runtime_src/driver/xclng/xrt/util/message.cpp [hunk 7]
```diff
@@ -74,18 +99,17 @@ class syslog_dispatch : public message_dispatch
 public:
   syslog_dispatch();
   virtual ~syslog_dispatch();
-  virtual void send(severity_level l, const char* msg) override;
+  virtual void send(severity_level l, const char* tag, const char* msg) override;
 private:
   std::map<severity_level, int> severityMap = {
```
Comment (sonals) [inline src/runtime_src/driver/xclng/xrt/util/message.cpp]: constexpr static std::map<severity_level, const char*> severityMap = {

File: src/runtime_src/driver/xclng/xrt/util/message.cpp [hunk 8]
```diff
@@ -96,19 +120,18 @@ class file_dispatch : public message_dispatch
   explicit
   file_dispatch(const std::string& file);
   virtual ~file_dispatch();
-  virtual void send(severity_level l, const char* msg) override;
+  virtual void send(severity_level l, const char* tag, const char* msg) override;
 private:
   std::ofstream handle;
   std::map<severity_level, const char*> severityMap = {
```
Comment (sonals) [inline src/runtime_src/driver/xclng/xrt/util/message.cpp]: constexpr static std::map<severity_level, const char*> severityMap = {

**Comment by AShivangi** [comment]
retest this please

**Comment by sonals** [comment]
retest this please

**Comment by sonals** [comment]
retest this please

**Pull request merged**